src/services/dashboards/dashboard-customize/modules/VariableSelectorDropdown.vue

```diff
@@ -95,18 +96,25 @@ const state = reactive({
             arrayOfSelectedOptions = option;
         } else arrayOfSelectedOptions = [option];
 
-        return arrayOfSelectedOptions.map((d) => ({ name: d, label: d }));
+        if (state.variableProperty.variable_type === 'MANAGED') {
+            return arrayOfSelectedOptions.map((d) => ({ name: d, label: props.referenceMap[d].label }));
+        } return arrayOfSelectedOptions.map((d) => ({ name: d, label: d }));
     }),
     options: computed<MenuItem[]>(() => {
-        const defaultOptions = state.variableSchema.options as string[];
-        return defaultOptions.map((d) => ({ name: d, label: d }));
+        let result;
+        if (state.variableProperty.variable_type === 'MANAGED') {
+            result = Object.keys(props.referenceMap).map((reference) => ({
+                name: props.referenceMap[reference].key, label: props.referenceMap[reference].label,
+            }));
```

Comment:
suggestion: for defensive code.

```suggestion
            result = Object.entries(props.referenceMap).map(([referenceKey, referenceItem]) => ({
                name: referenceKey,
                label: referenceItem?.label ?? referenceItem?.name ?? referenceKey,
            }));
```